### DISCUSSION TITLE
help with performance configuration

### DISCUSSION THREAD

**Discussion by bmeirellesRJ** [DNSCrypt/dnscrypt-proxy#2395] [Q&A]
in the example file, these 2 lines are commented out:

 dnscrypt_ephemeral_keys = false
tls_disable_session_tickets = false

If I activate the 2 lines, will I have a performance benefit?
What happens if I enable it?
Thanks

**Comment by lifenjoiner** [comment]
No, in short.

The defaults are both false. Commenting out doesn't really take effect. I think that's why the descriptions say "Only enable if ...":
https://github.com/DNSCrypt/dnscrypt-proxy/blob/9f86ffdd1e059915799a412198b4eb89f11d12ef/dnscrypt-proxy/example-dnscrypt-proxy.toml#L198-L207

`enable` means `true`.
The defaults are lower CPU consuming.
Enabling them increases privacy with extra calculations.

**Reply by bmeirellesRJ** [reply]
OK thank you

**Reply by flawrist** [reply]
But how an ordinary user can measure the increase in CPU load when `dnscrypt_ephemeral_keys` is set to `true`? I open the task manager (e.g. [Process Explorer](https://superuser.com/questions/1829513/what-is-the-best-way-to-identify-what-caused-a-cpu-spike)) and watch the CPU graph next to `dnscrypt-proxy.exe `while actively using the browser, but  I don't observe any noticeable spikes. Maybe there is a test for this case?
